TestProblem Solution Updated.py

```diff
@@ -0,0 +1,79 @@
+"""
+Dylan Williams
+Affinna Technical Part 2
+https://github.com/Lprods01/TestProblem
+March, 2024
+"""
+
+
+"""
+INPUT to this program:
+    > a series of .csv files
+OUTPUT of this program:
+    > a .txt file containing the total Resale Value of every entry
+    > a .txt file containing a dict with the total Resale Value of each unique item
+    > for each unique item, a .txt file containing that specific item's total Resale Value
+"""
+
+import csv
+import os
+
+inDir = "INPUT/"
+outDir = "OUTPUT/"
+if not os.path.exists(inDir):
+    os.makedirs(inDir)
+if not os.path.exists(outDir):
+    os.makedirs(outDir)
+
+
+
+
+for file in os.listdir(inDir):
+    
+    with open(inDir+file) as csvfile:
+        reader = csv.reader(csvfile)
+        next(reader)    #Skip header line containing field/column names. 
+    
+        totalResaleValue = 0
+        itemResVals = {}
+        for row in reader:
+            resVal = row[4]
+            resVal = resVal.replace('$', '')
+            resVal = resVal.replace(',', '')
+            resVal = float(resVal)
+
+            item = row[1]
+            if item in itemResVals:
+                itemResVals[item] += resVal
+            else:
+                itemResVals[item] = resVal
+
+            totalResaleValue += float(resVal)
+
+
+
+
+        #Write results to output files.
+        #.txt is used, though this portion can be modified to output .csv files.
+
+        #Each csv file in INPUT will have its own folder in OUTPUT.
+        outFileDir = outDir + file + "/"
+        if not os.path.exists(outFileDir):
+            os.makedirs(outFileDir)
+            
+        with open(outFileDir + "TotalResaleValue.txt", 'w') as outFile:
+            outFile.write(str(totalResaleValue))
+            
+        #The "Recommendations" section doesn't specify whether (1) all items should be in one file or (2) each item should have its own file.
+        #Both solutions are provided.        
```

Comment:
Which of the two solutions do you think is best for a real-world application? Likewise, what considerations might influence the decision to use one approach over the other?

Reply:
If another program needs to use these outputs, it would potentially be easier to fetch these totals from one file than from several files. With multiple files, there would be more overhead, as you would have to open, read, and close potentially hundreds/thousands of files, which is less robust. 

Outputting all items to one file could also make the project directory less cluttered, especially if there are hundreds/thousands of item types--there would be hundred or thousands of files in OUTPUT/.

To quickly find a total, it might be easier/faster to open the one output file and ctrl-f to find your desired item, as opposed to scouring through OUTPUT/.

However, storing all the calculations in one file could be a problem if the file becomes corrupted or is irretrievably deleted and the original inputs are not available to feed into the program again. You would be “putting all your eggs in one basket”, so to speak. 

These are the considerations that jump to mind. It seems that outputting to one well-organized and easily-processed file would be best.